addons/purchase_mrp/models/purchase_mrp.py

```diff
@@ -18,7 +18,7 @@ def _compute_qty_received(self):
         kit_lines = self.env['purchase.order.line']
         for line in self:
             if line.qty_received_method == 'stock_moves' and line.move_ids:
-                kit_bom = self.env['mrp.bom']._bom_find(product=line.product_id, company_id=line.company_id.id, bom_type='phantom')
+                kit_bom = self.env['mrp.bom'].with_context(active_test=True)._bom_find(product=line.product_id, company_id=line.company_id.id, bom_type='phantom')
```

Comment:
I would be dangerous if the user really archive its BoM and the stock.move contains the product correctly

Reply:
What's weird is that `self.env['mrp.bom']._bom_find(product=line.product_id, company_id=line.company_id.id, bom_type='phantom')` always returns an empty recordset (so, no archived BoM), except when Inventory Valuation is set to Automated.